staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -243,142 +300,337 @@ func TestRoundTripAndNewConnection(t *testing.T) {
+							StatusCode:       testCase.serverStatusCode,
+							ConnectionHeader: testCase.serverConnectionHeader,
+							UpgradeHeader:    "SPDY/3.1",
+						},
+					))
+					defer server.Close()
+
+					req, err := http.NewRequest("GET", server.URL, nil)
+					if err != nil {
+						t.Fatalf("%s: Error creating request: %s", k, err)
+					}
+
+					spdyTransport := NewRoundTripper(nil, redirect, redirect)
+
+					var proxierCalled bool
+					var proxyCalledWithHost string
+
+					interceptor := &Interceptor{proxyCalledWithHost: &proxyCalledWithHost}
+
+					proxyHandler := socks5Server(nil, interceptor)
+
+					if testCase.proxyAuth != nil {
+						proxyHandler = socks5Server(&socks5.StaticCredentials{
+							"proxyuser": "proxypasswd",
+						}, interceptor)
+					}
+
+					closed := make(chan struct{}, 1)
+					isClosed := func() bool {
+						select {
+						case <-closed:
+							return true
+						default:
+							return false
+						}
+					}
 
-				var proxierCalled bool
-				var proxyCalledWithHost string
-				var proxyCalledWithAuth bool
-				var proxyCalledWithAuthHeader string
-				if testCase.proxyServerFunc != nil {
-					proxyHandler := goproxy.NewProxyHttpServer()
+					l, err := net.Listen("tcp", "127.0.0.1:0")
+					if err != nil {
+						t.Fatalf("socks5Server: proxy_test: Listen: %v", err)
+					}
+					defer func() {
+						closed <- struct{}{}
+						l.Close()
+					}()
 
-					proxyHandler.OnRequest().HandleConnectFunc(func(host string, ctx *goproxy.ProxyCtx) (*goproxy.ConnectAction, string) {
-						proxyCalledWithHost = host
+					go func() {
+						conn, err := l.Accept()
+						if err != nil {
+							if isClosed() {
+								return
+							}
 
-						proxyAuthHeaderName := "Proxy-Authorization"
-						_, proxyCalledWithAuth = ctx.Req.Header[proxyAuthHeaderName]
-						proxyCalledWithAuthHeader = ctx.Req.Header.Get(proxyAuthHeaderName)
-						return goproxy.OkConnect, host
-					})
+							t.Errorf("error accepting connection: %s", err)
+						}
 
-					proxy := testCase.proxyServerFunc(proxyHandler)
+						if err := proxyHandler.ServeConn(conn); err != nil && !testCase.shouldError {
+							if isClosed() {
+								return
+							}
 
+							t.Errorf("ServeConn error: %s", err)
+						}
+					}()
 					spdyTransport.proxier = func(proxierReq *http.Request) (*url.URL, error) {
 						proxierCalled = true
-						proxyURL, err := url.Parse(proxy.URL)
-						if err != nil {
-							return nil, err
+						return &url.URL{
+							Scheme: "socks5",
+							Host:   net.JoinHostPort("127.0.0.1", strconv.Itoa(l.Addr().(*net.TCPAddr).Port)),
+							User:   testCase.proxyAuth,
+						}, nil
+					}
+
+					client := &http.Client{Transport: spdyTransport}
+
+					checkShouldError := func(err error) bool {
+						switch {
+						case err != nil && testCase.shouldError:
+							return true
+						case err == nil && testCase.shouldError:
+							t.Fatalf("%s: test should have raised an error but didn't", k)
```

Comment:
nit: no need to print "k" in the error message anymore.

There are a few references to `k` in this test. Can you remove all of them?

Reply:
done